frontends/web/src/utils/event-legacy.ts

```diff
@@ -15,61 +15,40 @@
  */
 
 import { apiWebsocket } from '../utils/websocket';
-
-/**
- * This type describes the subject of an event.
- */
-
-type Subject = string;
-
-/**
- * This interface models the legacy events that are received from the backend.
- */
-
-export interface IEvent {
-    readonly data?: Subject;
-    readonly deviceID?: string;
-    readonly meta?: null;
-    readonly type?: string;
-    readonly code?: string;
-}
+import { TEventLegacy, TPayload, TSubject } from '../api/socket';
 
 /**
  * This type describes the function used to observe the events.
  */
-
- type Observer = (event: IEvent) => void;
+type Observer = (event: TEventLegacy) => void;
 
 /**
  * This type describes the method returned to unsubscribe again.
  */
-
 export type IUnsubscribe = () => void;
 
 /**
  * This interface describes how the subscriptions are stored.
  */
-
 interface ISubscriptions {
     [subject: string]: Observer[];
 }
 
 /**
  * Stores the subscriptions as an object-based hash map.
  */
-
 const subscriptions: ISubscriptions = {};
 
 /**
  * This function dispatches the events from the websocket to the observers.
  */
-
-function handleMessages(payload: IEvent): void {
+function handleMessages(payload: TPayload): void {
```

Comment:
not `TEventLegacy`?

Reply:
handleMessages receives both `type TPayload = TEventLegacy | TEvent;`

and then later knows if it is `TEvent` or `TEventLegacy`. I'll make another comment further down where `observer(payload);` is called.